rust/lance-io/src/object_reader.rs

```diff
@@ -136,7 +136,7 @@ impl Reader for CloudObjectReader {
                 let meta = self
                     .do_with_retry(|| self.object_store.head(&self.path))
                     .await?;
-                Ok(meta.size)
+                Ok(meta.size as usize)
```

Comment:
The Lance Reader still requires usize. 

```
    /// Object/File Size.
    async fn size(&self) -> object_store::Result<usize>;

    /// Read a range of bytes from the object.
    ///
    /// TODO: change to read_at()?
    async fn get_range(&self, range: Range<usize>) -> object_store::Result<Bytes>;
```

It might be worth discussing if we should also change the trait to use u64. Let me know if we want to do that I can change in this PR as well.

Reply:
Yeah we need to do a refactor at some point to fix all the integers. There's a lot of place where we use `usize` where we shouldn't.